plugins/syntax/float_syntax.ml

```diff
@@ -22,8 +22,56 @@ let make_path dir id = Libnames.make_path (make_dir dir) (Id.of_string id)
 
 (*** Parsing for float in digital notation ***)
 
+let warn_inexact_float =
+  CWarnings.create ~name:"inexact-float" ~category:"parsing"
+    (fun (sn, f) ->
+      Pp.strbrk
+        (Printf.sprintf
+           "The constant %s is not a binary64 floating-point value. \
+            A closest value will be used and unambiguously printed %s."
+           sn (Float64.to_string f)))
```

Comment:
Apparently the message should be tuned to cope with the way infinities are really displayed in Coq:

```coq
Require Import Floats. Open Scope float_scope.
Check 1e1000.

> Check 1e1000.
>       ^^^^^^
Warning: The constant 1e1000 is not a binary64 floating-point value. A
closest value will be used and unambiguously printed inf.
[inexact-float,parsing]
infinity
     : float
```

and

```coq
Check -1e1000.

> Check -1e1000.
>       ^^^^^^^
Warning: The constant -1e1000 is not a binary64 floating-point value. A
closest value will be used and unambiguously printed -inf.
[inexact-float,parsing]
neg_infinity
     : float
```

But NaNs (especially the need to mask their sign) is not a issue here, given that a decimal literal won't be a NaN!

Reply:
fixed, please check though

Reply:
LGTM, thanks :+1: